tests/internal/e2e/test/specs/vcast_coded_test_completion.test.ts

```diff
@@ -0,0 +1,376 @@
+
+      for (const vcastTestingViewContentSection of await vcastTestingViewContent.getSections()) {
+        console.log(await vcastTestingViewContentSection.getTitle());
+        await vcastTestingViewContentSection.expand();
+        subprogram = await findSubprogram(
+          "manager",
+          vcastTestingViewContentSection
+        );
+        if (subprogram) {
+          if (!(await subprogram.isExpanded())) await subprogram.expand();
+          break;
+        }
+      }
+    }
+
+    if (!subprogram) {
+      throw "Subprogram 'manager' not found";
+    }
+
+    console.log("Looking for coded tests");
+    const subprogramMethod = await findSubprogramMethod(
+      subprogram,
+      "Coded Tests"
+    );
+    if (!subprogramMethod) {
+      throw "Subprogram method 'Coded Tests' not found";
+    }
+
+    if (!subprogramMethod.isExpanded()) {
+      await subprogramMethod.select();
+    }
+
+    let contextMenu = await subprogramMethod.openContextMenu();
+    console.log("Generating template test");
+    await contextMenu.select("VectorCAST");
+    let menuElement = await $("aria/Generate New Coded Test File");
+    await menuElement.click();
+
+    await (await $("aria/Save Code Test File")).click();
+    for (const character of "TestFiles/manager-template.cpp") {
+      await browser.keys(character);
+    }
+
+    await browser.keys(Key.Enter);
+
+    await bottomBar.openOutputView();
+    console.log("Checking that tests got generated");
+    let testHandle = await getTestHandle(
+      subprogram,
+      "Coded Tests",
+      "managerTests.ExampleFixtureTestCase",
+      2
+    );
+    expect(testHandle).not.toBe(undefined);
+
+    testHandle = await getTestHandle(
+      subprogram,
+      "Coded Tests",
+      "managerTests.ExampleTestCase",
+      2
+    );
+    expect(testHandle).not.toBe(undefined);
+
+    contextMenu = await testHandle.openContextMenu();
+    await contextMenu.select("VectorCAST");
+    menuElement = await $("aria/Edit Coded Test");
+    await menuElement.click();
+
+    const editorView = workbench.getEditorView();
+    await browser.waitUntil(
+      async () =>
+        (await (await editorView.getActiveTab()).getTitle()) ===
+        "manager-template.cpp"
+    );
+
+    const tab = (await editorView.openEditor(
+      "manager-template.cpp"
+    )) as TextEditor;
+    await browser.keys([Key.Ctrl, Key.Space]);
+    const contentAssist = await tab.toggleContentAssist(true);
+
+    // Check autocompletion for vsession
+    await tab.setTextAtLine(14, "vsessio");
+    let currentLine = await tab.getLineOfText("vsessio");
+    await tab.typeTextAt(currentLine, "vsessio".length + 1, "n");
+    await tab.save();
+    await browser.waitUntil(
+      async () => (await contentAssist.getItems()).length > 0
+    );
+
+    // Validating content assist for 'vsession'
+    expect(await contentAssist.hasItem("vsession")).toBe(true);
+    await selectItem(contentAssist, "vsession");
```

Comment:
This should be checking `auto vmock_session = ::vunit::MockSession();`, right?

Reply:
Added in 0dd0b1cff2d91e5bcfb6d7449c49b4b38bcb6c3a .